src/de/mod.rs

```diff
@@ -1,6 +1,1324 @@
+//!   - [`<element>` lists](#element-lists)
+//!
+//!
+//!
+//! Mapping XML to Rust types
+//! =========================
+//!
+//! Type names are never considered when deserializing, so you could name your
+//! types as you wish. Other general rules:
+//! - `struct` field name could be represented in XML only as an attribute name
+//!   or an element name;
+//! - `enum` variant name could be represented in XML only as an attribute name
+//!   or an element name;
+//! - the unit struct, unit type `()` and unit enum variant can be deserialized
+//!   from any valid XML content:
+//!   - attribute and element names;
+//!   - attribute and element values;
+//!   - text or CDATA content (including mixed text + CDATA content).
+//!
+//! <div style="background:rgba(120,145,255,0.45);padding:0.75em;">
+//!
+//! NOTE: examples, marked with `FIXME:` does not work yet -- any PRs that fixes
+//! that are welcome! The message after marker is a test failure message.
+//! Also, all that tests are marked with an `ignore` option, although their
+//! compiles. This is by intention, because rustdoc marks such blocks with
+//! an exclamation mark unlike `no_run` blocks.
+//!
+//! </div>
+//!
+//! <table>
+//! <thead>
+//! <tr><th>To parse all these XML's...</th><th>...use that Rust type(s)</th></tr>
+//! </thead>
+//! <tbody>
+//! <tr>
+//! <td>
+//! Content of attributes and text/CDATA content of elements (including mixed
+//! text and CDATA content):
+//!
+//! ```xml
+//! <... ...="content" />
+//! ```
+//! ```xml
+//! <...>content</...>
+//! ```
+//! ```xml
+//! <...><![CDATA[content]]></...>
+//! ```
+//! ```xml
+//! <...>text<![CDATA[cdata]]>text</...>
+//! ```
+//! </td>
+//! <td>
+//!
+//! You can use any type that can be deserialized from an `&str`, for example:
+//! - [`String`] and [`&str`]
+//! - [`u32`], [`f32`] and other numeric types
+//! - `enum`s, like
+//!   ```ignore
+//!   // FIXME: #474, merging mixed text / CDATA
+//!   // content does not work yet
+//!   # use pretty_assertions::assert_eq;
+//!   # use serde::Deserialize;
+//!   # #[derive(Debug, PartialEq)]
+//!   #[derive(Deserialize)]
+//!   enum Language {
+//!     Rust,
+//!     Cpp,
+//!     #[serde(other)]
+//!     Other,
+//!   }
+//!   # #[derive(Debug, PartialEq, Deserialize)]
+//!   # struct X { #[serde(rename = "$text")] x: Language }
+//!   # assert_eq!(X { x: Language::Rust  }, quick_xml::de::from_str("<x>Rust</x>").unwrap());
+//!   # assert_eq!(X { x: Language::Cpp   }, quick_xml::de::from_str("<x>C<![CDATA[p]]>p</x>").unwrap());
+//!   # assert_eq!(X { x: Language::Other }, quick_xml::de::from_str("<x><![CDATA[other]]></x>").unwrap());
+//!   ```
+//!
+//! <div style="background:rgba(120,145,255,0.45);padding:0.75em;">
+//!
+//! NOTE: deserialization to non-owned types (i.e. borrow from the input),
+//! such as `&str`, is possible only if you parse document in the UTF-8
+//! encoding and content do not contains XML escape sequences, such as `&lt;`
```

Comment:
```suggestion
//! encoding and content does not contain XML escape sequences such as `&lt;`,
```

`&lt;` is also an entity reference, it is just predefined by the specification. We should also mention character references. 

 You could say "does not contain entity references such as `&amp;`, or character references such as `&#xD;`.

Reply:
Rephrased as your last suggestion.